### PR TITLE
[Pre-training]  Optional for nsp task, support multi-datasets in ERNIE-1.0 pre-training

### PR DESCRIPTION
**Pull request by ZHUI**
### PR types

New features
### PR changes

Models
### Description

- Optional for nsp task,
- support multi-datasets

#### 效果对比 RoBERTa-wwm-ext-large

|- | RoBERTa-wwm-ext-large | ernie-1.0-ext-large
|-- | -- | --|
Masking | WWM | WWM
Type | large | large
Data Source | wiki+ext | clue + ernie + wudao
Training Tokens # | 5.4B | 4B/4B/ > 20B
Device | TPU Pod v3-32[3] | A100
Training Steps | 2M_MAX512 | 2M_MAX512
Batch Size | 512 | 512
Optimizer | AdamW | AdamW
Vocabulary | ~BERT | ~ernie-1.0
Init Checkpoint | Random Init | Random Init
NSP| False | False

1] EXT数据包括：中文维基百科，其他百科、新闻、问答等数据，总词数达5.4B

训练配置：batch size = 512, steps=200w
大致效果：
Model | Arch | CLUE AVG | 除WSC AVG | 
-- | -- | -- | -- | 
ERINE-1.0-ext-large | 24L1024H | 77.31(+1.19) | 74.68(+0.87) | 
RoBERTa-wwm-ext-large | 24L1024H | 76.14 | 73.81 |

### CODE REVIEW

File: examples/language_model/data_tools/dataset_utils.py [hunk 1]
```diff
@@ -38,6 +37,103 @@
...
+    import subprocess
+    path = os.path.abspath(os.path.dirname(__file__))
+    ret = subprocess.run(['make', '-C', path])
+    if ret.returncode != 0:
+        print("Making C++ dataset helpers module failed, exiting.")
+        import sys
+        sys.exit(1)
+
+
+class BlendableDataset(paddle.io.Dataset):
```
Comment (ZeyuChen) [inline examples/language_model/data_tools/dataset_utils.py:57]: Please add more comments, why it's called Blendable

Comment (ZHUI) [reply examples/language_model/data_tools/dataset_utils.py:57]: The BlendableDataset is a wrapper which used to mix different datasets.
The input is a list of datasets and corresponding weights for each dataset.

`Blendable` means the datasets are combined or mixed together.

File: paddlenlp/transformers/ernie/modeling.py [hunk 2]
```diff
@@ -132,6 +132,19 @@ class ErniePretrainedModel(PretrainedModel):
             "vocab_size": 18000,
             "pad_token_id": 0,
         },
+        "ernie-1.0-large": {
```
Comment (wawltor) [inline paddlenlp/transformers/ernie/modeling.py]: 有个疑问，因为这里为了预训练所以加了一个这样的ernie-1.0-large，但是没有加上ernie-1.0-large参数，这块需要讨论一下，避免用户错误使用

File: examples/language_model/ernie-1.0/run_pretrain_static.py [hunk 3]
```diff
@@ -61,12 +61,14 @@ def create_pretrained_dataset(
...
         places,
         data_holders,
+        binary_head=True,
         current_step=0, ):
 
     train_valid_test_num_samples = [
-        args.global_batch_size * args.max_steps, args.micro_batch_size *
-        (args.max_steps // args.eval_freq + 1) * args.eval_iters *
-        data_world_size, args.micro_batch_size * args.test_iters
+        args.global_batch_size * args.max_steps,
```
Comment (wawltor) [inline examples/language_model/ernie-1.0/run_pretrain_static.py:68]: 动态图的修改是不是也要加上binary_head

File: examples/language_model/data_tools/dataset_utils.py [hunk 4]
```diff
@@ -42,6 +41,109 @@ def get_local_rank():
...
+    prefixes = [0] * num_datasets
+    for i in range(num_datasets):
+        weights[i] = float(data_prefix[2 * i])
+        prefixes[i] = (data_prefix[2 * i + 1]).strip()
+    # Normalize weights
+    weight_sum = 0.0
+    for weight in weights:
+        weight_sum += weight
+    assert weight_sum > 0.0
+    weights = [weight / weight_sum for weight in weights]
```
Comment (wawltor) [inline examples/language_model/data_tools/dataset_utils.py:132]: 没有太明白这里的weight是什么weight，具体的作用是什么？

File: examples/language_model/ernie-1.0/run_pretrain_static.py (hunk 3 above)
Comment (ZHUI) [reply examples/language_model/ernie-1.0/run_pretrain_static.py:68]: 是的，修改后也可以支持。binary_head 为Fasle就没有了NSP

File: examples/language_model/data_tools/dataset_utils.py (hunk 4 above)
Comment (ZHUI) [reply examples/language_model/data_tools/dataset_utils.py:132]: 不同数据集的权重。比如，数据集A是4G，B是8G， 为了保证两者采样平衡，可以设置权重为 1:2，保证均匀采样。

File: paddlenlp/transformers/ernie/modeling.py (hunk 2 above)
Comment (ZHUI) [reply paddlenlp/transformers/ernie/modeling.py]: 可以删除。这个有计划添加上去，目前也不暴露给用户。

**Review by wawltor — APPROVED** [review]
LGTM
